examples/Mediawiki/MakeUser.yaml

```diff
@@ -0,0 +1,95 @@
+# This demonstrates the "resource" feature, similar to fixtures.
+# See T228889
+
+resource: User
+description: Creates a user and logs them in
+type: RPC
+
+use-fixtures:
+  - RootUser
+
+variables:
+  name: p/uniq: "User_"
+  email: p/uniq: "bob@example"
+  password: p/uniq: "pw-"
+
+setup:
+  # Get a createaccount token for the root user
+  - use-session: RootUser # use the root user's session, not the session created above
+    request:
+      path: api.php
+      parameters:
+        action: query
+        meta: tokens
+        type: createaccount
+        format: json
+    response:
+      body:
+        query:
+          tokens:
+            createaccount: p/grab: token # grab the value and store is as a local variable
+
+  # Create a new user account
+  - use-session: RootUser # use the root user's session, not the session created above
+    request:
+      path: api.php
+      parameters:
+        action: createaccount
+        format: json
+      form-data:
+        username: p/var: name
+        password: p/var: password
+        retype: p/var: password
+        email: p/var: email
+        creationtoken: p/var: token # from above
+    response:
+      body:
+        creataccount:
+          status: PASS
+
+  # Get a login token (using the new session created above)
+  - request:
+      path: api.php
+      parameters:
+        action: query
+        meta: tokens
+        type: login
+        format: json
+    response:
+      body:
+        query:
+          tokens:
+            login: p/grab: token # grab the value and store is as a local variable
+
+  # Log in (using the new session created above)
+  - request:
+      method: post
+      path: api.php
+      parameters:
+        action: login
+        format: json
+      form-data:
+        lgname: p/var: name
+        lgpassword: p/var: password
+        lgtoken: p/var: token
+    response:
+      body:
+        login:
+          result: "Success"
+          lguserid: p/grab: id
+          lgusername: p/grab: name
```

Comment:
Are these variables automatically exported or should a resource explicitly list the variables to be exported like fixtures?

Reply:
Either both should explicitly export, or neither should. I' kind of on the fence about that. What do you think?

Reply:
I think we should be consistent and explicitly export them.